scripts/target.sh

```diff
@@ -175,9 +210,43 @@ download_github_release() {
     cd - > /dev/null
 }
 
+# Function to get the correct file for a target and architecture
+get_target_file() {
+    local target=$1
+    local arch=$2
+    local files="${TARGETS[$target.file]}"
+    
+    if [ -z "$files" ]; then
+        echo ""
+        return 0
+    fi
+    
+    # Parse architecture-specific files
+    for file_spec in $files; do
+        if [[ "$file_spec" == "$arch:"* ]]; then
+            echo "${file_spec#*:}"
+            return 0
+        fi
+    done
+    
+    # If requested arch not found, return empty (let caller handle the error)
+    echo ""
+    return 1
+}
+
 run() {
     local target="$1"
-    local command="${TARGETS[$target.cmd]}"
+    local arch="${2:-linux}"
```

Comment:
let's pick the olatform to use from some bash command without requiring the use to exolicitly specify it